r/tests/testthat/test-dataset.R

```diff
@@ -194,6 +194,20 @@ test_that("Hive partitioning", {
   )
 })
 
+test_that("Hive partitioning without explicit argument name", {
```

Comment:
Nice test! We'll have to change this with the adjustment above, and when we do We should also include in the name (or possibly a comment) that the input we're giving is malformed, specifically that a partition is being passed to the schema argument.